src/ConfigInterface.php

```diff
@@ -26,7 +26,7 @@
  * @psalm-type FactoriesConfigurationType = array<
  *      string,
  *      (class-string<Factory\FactoryInterface>|Factory\FactoryInterface)
- *      |callable(ContainerInterface,string,array<mixed>|null)
+ *      |callable(ContainerInterface,string,array<mixed>|null):object
```

Comment:
This `object` return type hint was missing and led to an exception being thrown (see https://github.com/slevomat/coding-standard/issues/1330). Adding the return type hint resolves that exception.

`object` was chosen as this is already reflected in other annotations, although I believe `mixed` would technically be "allowed". Noting this for visibility.